File: 2022/AoC_6/day_6.js
const fs = require('fs');
const fileContent = fs.readFileSync('input_ex.txt', 'utf8');
const data = fileContent;
console.log(data)

let notFound = true
while (notFound) {
    for (let i=0; i<data.length; i++) {
        let charBox = data.slice(i,(i+4))
        // console.log(charBox)
        if (charBox.length === new Set(charBox).size) {
            console.log(charBox)
            idx = data.indexOf(charBox)+14
            console.log(idx)
            notFound = false
            break
        }
    }
}

File: 2022/AoC_6/day_6.py
with open("input_ex.txt") as file:
    data=file.read()

print(data)

i = 0
found = False
while not found:

    for i in range(len(data)):
        chars_box = data[i:i+14]
        #check if next 4 chars are different
        if len(chars_box) == len(set(chars_box)):
            print(chars_box)
            idx = data.find(chars_box) +14
            print(idx)
            found = True
            break
               
